sources/engine/Xenko.Engine/Engine/Processors/ScriptSystem.cs

```diff
@@ -127,7 +128,7 @@ public override void Update(GameTime gameTime)
         /// Allows to wait for next frame.
         /// </summary>
         /// <returns>ChannelMicroThreadAwaiter&lt;System.Int32&gt;.</returns>
-        public ChannelMicroThreadAwaiter<int> NextFrame()
+        public virtual ChannelMicroThreadAwaiter<int> NextFrame()
```

Comment:
Virtual methods are more expensive than non-virtual and can prevent inlining.

While making `Add` and `Remove` virtual will probably have a low impact, `NextFrame()` is called very often, hence its impact is non-trivial.